### ISSUE TITLE
NegatedIsNil not warning for guards with and/or

### ISSUE THREAD

**Issue by SamuelWillis** [rrrene/credo#978]
### Precheck

* Proposals for new features should be submitted via: https://github.com/rrrene/credo-proposals
* For bugs, please do a quick search and make sure the bug has not yet been reported here

### Environment

* Credo version (`mix credo -v`): 1.6.4
* Erlang/Elixir version (`elixir -v`): Elixir 1.13.4 (compiled with Erlang/OTP 24)
* Operating system: macOS Big Sur

### What were you trying to do?

Provide warnings when `not is_nil` is used in our codebase.

When a guard is written that is not just a `not is_nil` check credo fails to warn.

Examples:
```elixir
defmodule GuardModule do
  defguard guard_with_and(struct) when not is_nil(struct) and struct.foo == :baz
  
  defguard guard_with_or(struct) when not is_nil(struct) or struct.foo == :baz
end
```

### Expected outcome

The guards above would generate a warning when running `mix credo` with the `NegatedIsNil` rule is enabled

### Actual outcome

No warnings are printed.

```bash
mix credo guard_module.ex
Checking 1 source file ...

Please report incorrect results: https://github.com/rrrene/credo/issues

Analysis took 6.7 seconds (0.2s to load, 6.5s running 70 checks on 1 file)
1 mods/funs, found no issues.

Use `mix credo explain` to explain issues, `mix credo --help` for options.
```

**Comment by antedeguemon** [comment]
This issue was fixed by #960 but hasn't been shipped to a release yet.

The following test succeeds when ran with the latest `master` branch code:

```elixir
test "negated is nil" do
  """
  defmodule GuardModule do
    defguard guard_with_and(struct) when not is_nil(struct) and struct.foo == :baz

    defguard guard_with_or(struct) when not is_nil(struct) or struct.foo == :baz
  end
  """
  |> to_source_file()
  |> run_check(Credo.Check.Refactor.NegatedIsNil)
  |> assert_issues()
end
```

**Comment by SamuelWillis** [comment]
My apologies! I should have checked the `master` branch. Thank you.

**Comment by antedeguemon** [comment]
@SamuelWillis No need to apologize! Thank you for the report 😄

**Comment by rrrene** [comment]
This is now live in ``v1.6.5` :+1:
